File: CodeChef/Practice/Jmarket.py
t=int(input())
while(t>0):
    d=input().split()
    x=int(d[0])
    a=int(d[1])
    b=int(d[2])
    c=int(d[3])
    mx=max(a,b,c)
    mn=min(a,b,c)
    md=a+b+c-mx-mn
    print(mn*(x-1)+md)

File: CodeChef/Practice/Jmarket.cpp
#include <iostream>
#include<bits/stdc++.h>
using namespace std;

int main(){
    int t;
    cin>>t;
    while(t--){
        int x;
        cin>>x;
        int a,b,c;
        cin>>a>>b>>c;
        int mn=min(a,min(b,c)); 
        int mx=max(a,max(b,c));
        int md=a+b+c-mn-mx;
        cout<<mn*(x-1)+md<<endl;
    }
    return 0;
}